packages/file/lib/src/backends/record_replay/common.dart

```diff
@@ -155,3 +156,29 @@ bool _areMapsEqual<K, V>(Map<K, V> map1, Map<K, V> map2) {
             deeplyEqual(map1[key], map2[key]);
       });
 }
+
+/// Returns a human-readable representation of an [Invocation].
+String describeInvocation(Invocation invocation) {
+  StringBuffer buf = StringBuffer();
+  buf.write(getSymbolName(invocation.memberName));
+  if (invocation.isMethod) {
+    buf.write('(');
+    int i = 0;
+    for (dynamic arg in invocation.positionalArguments) {
+      if (i++ > 0) {
```

Comment:
here and below: flutter style says not to use ++ operator, particularly in this kind of situation where it's harder to immediately see what you're actually comparing here. Prefer putting `i += 1` at the end of this, or maybe just converting it to a regular `for` loop.

Alternatively, you could get rid of the variable entirely and just use `invocation.positionalArguments.skip(1)`, after first writing the start of the buffer right?

Reply:
This change is almost entirely moving existing code around, and I was trying to reduce noise in the diff.  I don't mind cleaning up `i++` and the variable names to match modern style, but I think I'd prefer to do it separately.

Regarding removing the variable: I think that would add a lot of extra complication.
1. Would need to check that `positionalArguments` is non-empty before handling the first element.
2. Handling the first element separately would duplicate code.
3. The variable is also used by the `namedArguments` loop, which similarly needs to know if any positional argument were already printed so that it can print a comma (or not).